util/redisutil/test_redis.go

```diff
@@ -1,21 +1,22 @@
 // Copyright 2021-2022, Offchain Labs, Inc.
 // For license information, see https://github.com/nitro/blob/master/LICENSE
 
-//go:build redistest
-// +build redistest
-
 package redisutil
 
 import (
-	"os"
+	"fmt"
 	"testing"
+
+	"github.com/alicebob/miniredis/v2"
+
+	"github.com/offchainlabs/nitro/util/testhelpers"
 )
 
 // t param is used to make sure this is only called in tests
-func GetTestRedisURL(t *testing.T) string {
-	redisUrl := os.Getenv("TEST_REDIS")
-	if redisUrl == "" {
-		redisUrl = DefaultTestRedisURL
-	}
-	return redisUrl
+func GetTestRedisURL(t *testing.T) (*miniredis.Miniredis, string) {
+	redisServer, err := miniredis.Run()
+	testhelpers.RequireImpl(t, err)
+
+	return redisServer, fmt.Sprintf("redis://%s/0", redisServer.Addr())
```

Comment:
we should also support using external redis. I don't trust miniredis with things like atomicity checks.
1 option is to launch miniredis if TEST_REDIS env variable is not set, and use external redis if it is.

Reply:
Done.